plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -823,34 +822,116 @@ protected function get_and_maybe_update_months_filter_cache(): array {
 			$current_year_month->year  = gmdate( 'Y', time() );
 			$current_year_month->month = gmdate( 'n', time() );
 			if ( count( $cached_value ) === 0 || (
-				$cached_value[0]->year !== $current_year_month->year ||
-				$cached_value[0]->month !== $current_year_month->month )
+					$cached_value[0]->year !== $current_year_month->year ||
+					$cached_value[0]->month !== $current_year_month->month )
 			) {
 				array_unshift( $cached_value, $current_year_month );
 			}
 			return $cached_value;
 		}
 
+		// Option to use a simplified, more performant method for building the months options list.
+		$use_simplified_months_options = boolval( get_option( 'wc_' . $this->order_type . '_list_table_months_filter_simplified', false ) );
+
+		if ( true === $use_simplified_months_options ) {
+			$order_dates = $this->get_simplified_months_filter_options();
+		} else {
+			$orders_table = esc_sql( OrdersTableDataStore::get_orders_table_name() );
+			$trash_status = esc_sql( OrderStatus::TRASH );
+			$utc_offset   = wc_timezone_offset();
+
+			$order_dates = $wpdb->get_results(
+				// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				$wpdb->prepare(
+					"
+						SELECT DISTINCT YEAR( t.date_created_local ) AS year,
+										MONTH( t.date_created_local ) AS month
+						FROM ( SELECT DATE_ADD( date_created_gmt, INTERVAL $utc_offset SECOND ) AS date_created_local FROM $orders_table WHERE type = %s AND status != '$trash_status' ) t
+						ORDER BY year DESC, month DESC
+					",
+					$this->order_type
+				)
+				// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			);
+		}
+
+		update_option( $cache_option_date_name, time(), false );
```

Comment:
I know this was copied over from the old approach, but curious why this uses option setters over `wp_cache` or even transients with expirations.